### ISSUE TITLE
Error when get focus

### ISSUE THREAD

**Issue by igorregis** [pinball83/Masked-Edittext#17]
I am getting the error below with the configuration as described

`````` xml
<com.github.pinball83.maskededittext.MaskedEditText
            android:id="@+id/textPrecoReais"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            android:layout_alignBaseline="@+id/textViewRCifrao"
            android:layout_toRightOf="@id/textViewRCifrao"
            android:ems="5"
            android:selectAllOnFocus="true"
            android:digits="@string/localized_digitis"
            android:inputType="number"
            app:mask="***.***,**"
            app:notMaskedSymbol="*"
            />

```java
 Process: br.com.cacs, PID: 16922
 java.lang.IndexOutOfBoundsException: charAt: 0 >= length 0
     at android.text.SpannableStringBuilder.charAt(SpannableStringBuilder.java:114)
     at com.github.pinball83.maskededittext.MaskedEditText.getUnmaskedText(MaskedEditText.java:198)
     at com.github.pinball83.maskededittext.MaskedEditText.onTouch(MaskedEditText.java:290)
     at android.view.View.dispatchTouchEvent(View.java:7708)
     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2216)
     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1917)
     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2216)
     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1917)
     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2216)
     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1917)
     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2216)
     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1917)
     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2216)
     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1917)
     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2216)
     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1917)
     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2216)
     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1917)
     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2216)
     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1917)
     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2216)
     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1917)
     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2216)
     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1917)
     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2216)
     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1917)
     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2216)
     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1917)
     at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2216)
     at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1917)
     at com.android.internal.policy.impl.PhoneWindow$DecorView.superDispatchTouchEvent(PhoneWindow.java:2329)
     at com.android.internal.policy.impl.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1568)
     at android.app.Activity.dispatchTouchEvent(Activity.java:2458)
     at android.support.v7.view.WindowCallbackWrapper.dispatchTouchEvent(WindowCallbackWrapper.java:63)
     at android.support.v7.view.WindowCallbackWrapper.dispatchTouchEvent(WindowCallbackWrapper.java:63)
     at com.android.internal.policy.impl.PhoneWindow$DecorView.dispatchTouchEvent(PhoneWindow.java:2277)
     at android.view.View.dispatchPointerEvent(View.java:7893)
     at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:3950)
     at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:3829)
     at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3395)
     at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3445)
     at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3414)
     at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:3521)
     at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:3422)
     at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:3578)
     at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3395)
     at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3445)
     at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3414)
     at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:3422)
     at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3395)
     at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:5535)
    at android.view.ViewRootImpl.doProc
``````

**Comment by igorregis** [comment]
I believe you just need to change tou getUnmaskedText to this:
Note the if statement [ if (text != null && text.length() > 0)  ]

``` Java
 public String getUnmaskedText() {
        Editable text = super.getText();
        if (mask != null && !mask.isEmpty()) {
            Editable unMaskedText = new SpannableStringBuilder();
            for (Integer index : listValidCursorPositions) {
                if (text != null && text.length() > 0) {
                    unMaskedText.append(text.charAt(index));
                }
            }
            if (format != null && !format.isEmpty())
                return formatText(unMaskedText.toString(), format);
            else
                return unMaskedText.toString().trim();
        }

        return text.toString().trim();
    }
```

**Comment by pinball83** [comment]
Fix merged in 1.0.4
